File: dmn-test-cases/standard/tck/1.3/cl3/0085-decision-services/translator/expected/java/dmn/DecisionService_007.java

import java.util.*;
import java.util.stream.Collectors;

@javax.annotation.Generated(value = {"ds.ftl", "decisionService_007"})
@com.gs.dmn.runtime.annotation.DRGElement(
    namespace = "",
    name = "decisionService_007",
    label = "",
    elementKind = com.gs.dmn.runtime.annotation.DRGElementKind.DECISION_SERVICE,
    expressionKind = com.gs.dmn.runtime.annotation.ExpressionKind.OTHER,
    hitPolicy = com.gs.dmn.runtime.annotation.HitPolicy.UNKNOWN,
    rulesCount = -1
)
public class DecisionService_007 extends com.gs.dmn.runtime.JavaTimeDMNBaseDecision {
    public static final com.gs.dmn.runtime.listener.DRGElement DRG_ELEMENT_METADATA = new com.gs.dmn.runtime.listener.DRGElement(
        "",
        "decisionService_007",
        "",
        com.gs.dmn.runtime.annotation.DRGElementKind.DECISION_SERVICE,
        com.gs.dmn.runtime.annotation.ExpressionKind.OTHER,
        com.gs.dmn.runtime.annotation.HitPolicy.UNKNOWN,
        -1
    );

    private static class DecisionService_007LazyHolder {
        static final DecisionService_007 INSTANCE = new DecisionService_007();
    }
    public static DecisionService_007 instance() {
        return DecisionService_007LazyHolder.INSTANCE;
    }

    private final Decision_007_2 decision_007_2;

    private DecisionService_007() {
        this(new Decision_007_2());
    }

    private DecisionService_007(Decision_007_2 decision_007_2) {
        this.decision_007_2 = decision_007_2;
    }

    @java.lang.Override()
    public Boolean applyMap(java.util.Map<String, String> input_, com.gs.dmn.runtime.ExecutionContext context_) {
        try {
            return apply(input_.get("decision_007_3"), context_);
        } catch (Exception e) {
            logError("Cannot apply decision 'DecisionService_007'", e);
            return null;
        }
    }

    public Boolean apply(String decision_007_3, com.gs.dmn.runtime.ExecutionContext context_) {
        try {
            // Start DS 'decisionService_007'
            com.gs.dmn.runtime.annotation.AnnotationSet annotationSet_ = context_ != null ? context_.getAnnotations() : null;
            com.gs.dmn.runtime.listener.EventListener eventListener_ = context_ != null ? context_.getEventListener() : null;
            com.gs.dmn.runtime.external.ExternalFunctionExecutor externalExecutor_ = context_ != null ? context_.getExternalFunctionExecutor() : null;
            com.gs.dmn.runtime.cache.Cache cache_ = context_ != null ? context_.getCache() : null;
            long decisionService_007StartTime_ = System.currentTimeMillis();
            com.gs.dmn.runtime.listener.Arguments decisionService_007Arguments_ = new com.gs.dmn.runtime.listener.Arguments();
            decisionService_007Arguments_.put("decision_007_3", decision_007_3);
            eventListener_.startDRGElement(DRG_ELEMENT_METADATA, decisionService_007Arguments_);

            // Bind input decisions
            cache_.bind("decision_007_3", decision_007_3);

            // Evaluate DS 'decisionService_007'
            Boolean output_ = lambda.apply(decision_007_3, context_);

            // End DS 'decisionService_007'
            eventListener_.endDRGElement(DRG_ELEMENT_METADATA, decisionService_007Arguments_, output_, (System.currentTimeMillis() - decisionService_007StartTime_));

            return output_;
        } catch (Exception e) {
            logError("Exception caught in 'decisionService_007' evaluation", e);
            return null;
        }
    }

    public com.gs.dmn.runtime.LambdaExpression<Boolean> lambda =
        new com.gs.dmn.runtime.LambdaExpression<Boolean>() {
            public Boolean apply(Object... args_) {
                String decision_007_3 = 0 < args_.length ? (String) args_[0] : null;
                com.gs.dmn.runtime.ExecutionContext context_ = 1 < args_.length ? (com.gs.dmn.runtime.ExecutionContext) args_[1] : null;
                com.gs.dmn.runtime.annotation.AnnotationSet annotationSet_ = context_ != null ? context_.getAnnotations() : null;
                com.gs.dmn.runtime.listener.EventListener eventListener_ = context_ != null ? context_.getEventListener() : null;
                com.gs.dmn.runtime.external.ExternalFunctionExecutor externalExecutor_ = context_ != null ? context_.getExternalFunctionExecutor() : null;
                com.gs.dmn.runtime.cache.Cache cache_ = context_ != null ? context_.getCache() : null;

                // Apply child decisions
                Boolean decision_007_2 = DecisionService_007.this.decision_007_2.apply(context_);

                return decision_007_2;
            }
    };
}

File: dmn-test-cases/standard/tck/1.3/cl3/0085-decision-services/translator/expected/python/dmn/DecisionService_007.py
import typing
import decimal
import datetime
import isodate
import time

import jdmn.runtime.Context
import jdmn.runtime.DefaultDMNBaseDecision
import jdmn.runtime.ExecutionContext
import jdmn.runtime.LambdaExpression
import jdmn.runtime.LazyEval
import jdmn.runtime.Pair
import jdmn.runtime.Range
import jdmn.runtime.RuleOutput
import jdmn.runtime.RuleOutputList

import jdmn.runtime.annotation.Annotation
import jdmn.runtime.annotation.AnnotationSet
import jdmn.runtime.annotation.DRGElementKind
import jdmn.runtime.annotation.ExpressionKind
import jdmn.runtime.annotation.HitPolicy

import jdmn.runtime.cache.Cache

import jdmn.runtime.external.ExternalFunctionExecutor

import jdmn.runtime.listener.Arguments
import jdmn.runtime.listener.DRGElement
import jdmn.runtime.listener.EventListener
import jdmn.runtime.listener.Rule

import Decision_007_2


# Generated(value = ["ds.ftl", "decisionService_007"])
class DecisionService_007(jdmn.runtime.DefaultDMNBaseDecision.DefaultDMNBaseDecision):
    DRG_ELEMENT_METADATA: jdmn.runtime.listener.DRGElement.DRGElement = jdmn.runtime.listener.DRGElement.DRGElement(
        "",
        "decisionService_007",
        "",
        jdmn.runtime.annotation.DRGElementKind.DRGElementKind.DECISION_SERVICE,
        jdmn.runtime.annotation.ExpressionKind.ExpressionKind.OTHER,
        jdmn.runtime.annotation.HitPolicy.HitPolicy.UNKNOWN,
        -1
    )
    _instance = None

    def __init__(self):
        raise RuntimeError("Call instance() instead")

    @classmethod
    def instance(cls):
        if cls._instance is None:
            cls._instance = cls.__new__(cls)
            jdmn.runtime.DefaultDMNBaseDecision.DefaultDMNBaseDecision.__init__(cls._instance)
            cls._instance.initSubDecisions()
        return cls._instance

    def initSubDecisions(self, decision_007_2: Decision_007_2.Decision_007_2 = None):
        self.decision_007_2 = Decision_007_2.Decision_007_2() if decision_007_2 is None else decision_007_2

    def apply(self, decision_007_3: typing.Optional[str], context_: jdmn.runtime.ExecutionContext.ExecutionContext) -> typing.Optional[bool]:
        try:
            # Start DS 'decisionService_007'
            annotationSet_: jdmn.runtime.annotation.AnnotationSet.AnnotationSet = None if context_ is None else context_.annotations
            eventListener_: jdmn.runtime.listener.EventListener.EventListener = None if context_ is None else context_.eventListener
            externalExecutor_: jdmn.runtime.external.ExternalFunctionExecutor.ExternalFunctionExecutor = None if context_ is None else context_.externalFunctionExecutor
            cache_: jdmn.runtime.cache.Cache.Cache = None if context_ is None else context_.cache
            decisionService_007StartTime_ = int(time.time_ns()/1000)
            decisionService_007Arguments_ = jdmn.runtime.listener.Arguments.Arguments()
            decisionService_007Arguments_.put("decision_007_3", decision_007_3)
            eventListener_.startDRGElement(self.DRG_ELEMENT_METADATA, decisionService_007Arguments_)

            # Bind input decisions
            cache_.bind("decision_007_3", decision_007_3)

            # Evaluate DS 'decisionService_007'
            output_: typing.Optional[bool] = self.evaluate(decision_007_3, context_)

            # End DS 'decisionService_007'
            eventListener_.endDRGElement(self.DRG_ELEMENT_METADATA, decisionService_007Arguments_, output_, (int(time.time_ns()/1000) - decisionService_007StartTime_))

            return output_
        except Exception as e:
            self.logError("Exception caught in 'decisionService_007' evaluation", e)
            return None

    def evaluate(self, decision_007_3: typing.Optional[str], context_: jdmn.runtime.ExecutionContext.ExecutionContext) -> typing.Optional[bool]:
        # Apply child decisions
        decision_007_2: typing.Optional[bool] = self.decision_007_2.apply(context_)

        return decision_007_2
